peps/pep-0789.rst

```diff
@@ -471,6 +468,32 @@ relate to this proposal, and to PEP 533 and PEP 568, then see `this comment
 <https://discuss.python.org/t/preventing-yield-inside-certain-context-managers/1091>`__.
 
 
+Deprecate async generators entirely
+-----------------------------------
+
+At the 2024 language summit, several attendees suggested instead deprecating async
+generators *in toto.*  I'm not opposed to that [#exp-report]_ -- it's a reasonable
+alternative to this PEP, albeit with enormous backwards-compatibilty costs.
+Two library features that I think would be important in this scenario are:
+
+1. Ergonomic async iterables in the standard library.  This could be addressed by
+   adding ``__aiter__`` to ``asyncio.Queue``, and a ``close()`` method to allow
```

Comment:
In 3.13 we've added [`.shutdown()`](https://docs.python.org/3.13/library/asyncio-queue.html#asyncio.Queue.shutdown) which I think takes care of `.close()` (unless the name is important? -- I'm not sure I understand why this is a prerequisite here) and `__aiter__` could then be done by subclassing?

Reply:
Yep, that will work!  I wasn't aware of shutdown; I'd still support aiter on the standard Queue class though.

Reply:
Adding `__aiter__` seems fairly uncontroversial now we have `shutdown`. It can just be discussed on Discourse. I'd recommend adding `__iter__` to non-asyncio `queues.Queue` as well, just like we added `shutdown` to both.